golem-service-base/src/worker_executor_clients.rs

```diff
@@ -1,24 +1,45 @@
+use crate::model::Pod;
 use async_trait::async_trait;
 use golem_api_grpc::proto::golem::workerexecutor::worker_executor_client::WorkerExecutorClient;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::Mutex;
 use tonic::transport::Channel;
 
-use crate::model::Pod;
+type WorkerExecutorCache = Arc<Mutex<HashMap<Pod, WorkerExecutorClient<Channel>>>>;
```

Comment:
You should use my cache implementation - it's async, supports different invalidation techniques and we use it for all caching things in the worker executor and the shard manager.

https://github.com/golemcloud/golem-services/blob/main/golem-common/src/cache.rs